src/classes/PMT_Payment_TEST.cls

```diff
@@ -257,6 +257,188 @@ private class PMT_Payment_TEST {
         System.assertEquals(closedWonStage, opp.StageName);
     }
 
+    /**
+     * @description Validate that the accounting rules are enforced properly. If there is a paid payment, it must have a payment date.
+     * If it is an unpaid payment, it must have a scheduled date. 
+     * down the Payment records.
+     */
+    static testMethod void test_EnforceAccountingDataConsistency() {
+        UTIL_CustomSettingsFacade.getContactsSettingsForTests(new npe01__Contacts_And_Orgs_Settings__c (npe01__Payments_Enabled__c= true, Enforce_Accounting_Data_Consistency__c = true));
+        Account acc = [SELECT Id FROM Account LIMIT 1];
+
+
+        Opportunity opp = new Opportunity(
+                Name = 'success',
+                Amount = 10,
+                AccountID = acc.id,
+                CloseDate = system.today(),
+                StageName = UTIL_UnitTestData_TEST.getClosedWonStage());
+        insert opp;
+
+        Test.startTest();
+
+        // ****************** Test Insertion *******************************
+        // Test the invalid paid insertion case 
+        npe01__OppPayment__c invalidPaidPayment = new npe01__oppPayment__c(npe01__Opportunity__c = opp.Id, npe01__Paid__c = true);
+        try {
+            insert invalidPaidPayment;
+        } catch (Exception e) {
+            System.assert(e.getMessage().contains('A paid payment must have a payment date.'), 'The error message is not expected, actual: ' + e.getMessage());
```

Comment:
When you change to Custom Labels, update here too.

Reply:
Updated.